api/eds.proto

```diff
@@ -74,10 +74,16 @@ message LbEndpoint {
   // Optional health status when known and supplied by EDS server.
   HealthStatus health_status = 2;
 
-  // The optional canary status of the upstream host. Envoy uses the canary
-  // status for various statistics and load balancing tasks documented
-  // elsewhere.
-  google.protobuf.BoolValue canary = 3;
+  // The endpoint metadata specifies values that may be used by the load
+  // balancer to select endpoints in a cluster for a given request. The filter
+  // name should be specified as "envoy.lb". An example boolean key-value pair
+  // is "canary", providing the optional canary status of the upstream host.
+  // This may be matched against in a route's ForwardAction metadata_match field
+  // to subset the endpoints considered in cluster load balancing.
+  // TODO(htuch: [V2-API-DIFF] Need to plumb this through Envoy and have
+  // everywhere that canary is used be capable of working on metadata.
+  Metadata metadata = 3;
+
```

Comment:
This can't co-exist with the weighted-clusters one can it? There would be two metadata sets (one at weighted cluster and one here).